pkg/services/ngalert/notifier/silence_svc.go

```diff
@@ -0,0 +1,127 @@
+package notifier
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/auth/identity"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+// SilenceService is the authenticated service for managing alertmanager silences.
+type SilenceService struct {
+	authz      SilenceAccessControlService
+	xact       transactionManager
+	log        log.Logger
+	stateStore SilenceStateStore
+}
+
+// SilenceAccessControlService provides access control for silences.
+type SilenceAccessControlService interface {
+	FilterByAccess(ctx context.Context, user identity.Requester, silences ...*models.Silence) ([]*models.Silence, error)
+	AuthorizeReadSilence(ctx context.Context, user identity.Requester, silence *models.Silence) error
+	AuthorizeCreateSilence(ctx context.Context, user identity.Requester, silence *models.Silence) error
+	AuthorizeUpdateSilence(ctx context.Context, user identity.Requester, silence *models.Silence) error
+}
+
+// SilenceStateStore is the interface for storing and retrieving silences. Currently, this is implemented by
+// MultiOrgAlertmanager but should eventually be replaced with an actual store.
+type SilenceStateStore interface {
```

Comment:
why do you use "state"? I think just silence store is clear enough

Reply:
27a08da83de23881ff47d1d918fd45f04139ced8